go/libraries/doltcore/sqle/dfunctions/hashof_database.go

```diff
@@ -0,0 +1,88 @@
+// Copyright 2024 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dfunctions
+
+import (
+	"fmt"
+
+	"github.com/dolthub/go-mysql-server/sql"
+	"github.com/dolthub/go-mysql-server/sql/types"
+
+	"github.com/dolthub/dolt/go/libraries/doltcore/sqle/dsess"
+)
+
+const HashOfDatabaseFuncName = "dolt_hashof_db"
+
+type HashOfDatabase struct{}
+
+// NewHashOfDatabase creates a new HashOfDatabase expression.
+func NewHashOfDatabase() sql.Expression {
+	return &HashOfDatabase{}
+}
+
+// Children implements the Expression interface.
+func (t *HashOfDatabase) Children() []sql.Expression {
+	return nil
+}
+
+// Eval implements the Expression interface.
+func (t *HashOfDatabase) Eval(ctx *sql.Context, row sql.Row) (interface{}, error) {
```

Comment:
Make this take 0 or 1 args

0 args, behaves as now

1 arg, treat it as the name of a ref (or the pseudo-ref 'WORKING', 'STAGED')